Code/GraphMol/RGroupDecomposition/RGroupCore.cpp

```diff
@@ -0,0 +1,365 @@
+
+#include "RGroupCore.h"
+
+namespace RDKit {
+
+namespace {
+// From answer 12 in
+// https://stackoverflow.com/questions/5279051/how-can-i-create-cartesian-product-of-vector-of-vectors
+// by anumi
+static std::vector<std::vector<int>> cartesianProduct(
+    const std::vector<std::vector<int>> &v) {
+  std::vector<std::vector<int>> s = {{}};
+  for (const auto &u : v) {
+    std::vector<std::vector<int>> r;
+    for (const auto &x : s) {
+      for (const auto y : u) {
+        r.push_back(x);
+        r.back().push_back(y);
+      }
+    }
+    s = move(r);
```

Comment:
I think this should be `std::move(r)`

Reply:
Changed- though I'm not sure how it worked originally!